src/Control/Lens/Prism.hs

```diff
@@ -361,3 +369,71 @@ _Show = prism show $ \s -> case reads s of
   [(a,"")] -> Right a
   _ -> Left s
 {-# INLINE _Show #-}
+
+class Prefixed t where
+  -- | A 'Prism' stripping a prefix from a sequence when used as a 'Traversal',
+  -- or prepending that prefix when run backwards:
+  --
+  -- >>> "preview" ^? prefixed "pre"
+  -- Just "view"
+  --
+  -- >>> "review" ^? prefixed "pre"
+  -- Nothing
+  --
+  -- >>> prefixed "pre" # "amble"
+  -- "preamble"
+  prefixed :: t -> Prism' t t
+
+instance Eq a => Prefixed [a] where
+  prefixed ps = prism' (ps ++) (List.stripPrefix ps)
+  {-# INLINE prefixed #-}
+
+instance Prefixed TS.Text where
+  prefixed p = prism' (p <>) (TS.stripPrefix p)
+  {-# INLINE prefixed #-}
+
+instance Prefixed TL.Text where
+  prefixed p = prism' (p <>) (TL.stripPrefix p)
+  {-# INLINE prefixed #-}
+
+instance Prefixed BS.ByteString where
+  prefixed p = prism' (p <>) (BS.stripPrefix p)
+  {-# INLINE prefixed #-}
+
+instance Prefixed BL.ByteString where
```

Comment:
Good. We have `Cons` instances for `ZipList`, `Seq`, various `Vector`s and lens-own `Deque`.

- ZipList should be easy to add
- `Seq` doen't have stripPrefix/stripSuffix, neither `Vector`, but they have `splitAt` and O(1) `length`, so should be  easy to add.
- I don't know `Deque`, so that could be left out.

Reply:
I do think we should do this, although I don't want it to hold up this PR. I've opened #989 as a reminder to do this.